pkg/storage/bucketstorage.go

```diff
@@ -0,0 +1,55 @@
+package storage
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// StorageProvider is the type of provider used for storage if not leveraging a file implementation.
+type StorageProvider string
+
+const (
+	S3 StorageProvider = "S3"
+	// AZURE StorageProvider = "AZURE"
+	// GCS   StorageProvider = "GCS"
+)
+
+// StorageConfig is the configuration type used to
+type StorageConfig struct {
+	Type   StorageProvider `yaml:"type"`
+	Config interface{}     `yaml:"config"`
```

Comment:
It has to be an interface because, as it's explained in the NewBucketStorage, which has been around and reviewed previously, stroage leverages a thanos configuration format: https://thanos.io/tip/thanos/storage.md/

It needs to be typed as an interface{} because it can be a variety of different objects. Note that it is marshalled again before being passed to the specific storage implementation.